main/src/main/java/cgeo/geocaching/connector/internal/InternalConnector.java

```diff
@@ -211,15 +213,14 @@ protected static int getUdcListId(final Context context) {
      * creates a new cache with the given id, if it does not exist yet
      *
      * @param context       context in which this function gets called
-     * @param id            internal (numeric) id of the cache
+     * @param geocode       internal id of the cache
      * @param name          cache's name (or null for default name)
      * @param description   cache's description (or null for default description)
      * @param assignedEmoji cache's assigned emoji (or 0 for default cache type icon)
      * @param geopoint      cache's current location (or null if none)
      * @param listId        cache list's id, may be NEW_LIST
      */
-    protected static void assertCacheExists(final Context context, final long id, @Nullable final String name, @Nullable final String description, final int assignedEmoji, @Nullable final Geopoint geopoint, final int listId) {
-        final String geocode = geocodeFromId(id);
+    protected static void assertCacheExists(final Context context, final String geocode, @Nullable final String name, @Nullable final String description, final int assignedEmoji, @Nullable final Geopoint geopoint, final int listId) {
```

Comment:
I believe method can be made private (this is unrelated to the issue but would be nice anyway :-))